src/lua/log.lua

```diff
@@ -56,110 +61,203 @@ local log_level_name = setmetatable({
+            return
+        end
+        options = fun.iter(options or {}):tomap()
+        -- level configuration
+        local level = options.level
+        if level ~= nil then
+            if type(options.level) == 'string' then
+                level = log_level_name[level]
+                if level == nil then
+                    error('bad option level (bad logging level)')
+                end
+            elseif type(level) == 'number' then
+                if level < S_SYSERROR or level > S_DEBUG then
+                    error('bad option level (bad logging level)')
+                end
+            else
+                error('bad option level (bad logging level)')
+            end
+            self:_set_level(level)
+        elseif not self.is_configured then
+            self:_set_level(S_INFO)
+        end
+        -- backtrace configuration
+        local backtrace_on_error = options.backtrace_on_error
+        if type(backtrace_on_error) == 'boolean' then
+            self.backtrace_on_error = backtrace_on_error
+        elseif not self.is_configured then
+            self.backtrace_on_error = false
+        end
+        self.is_configured = true
+    end,
+    warn = function(self, fmt, ...)
+        self:_log(nil, S_WARN, fmt, ...)
+    end,
+    info = function(self, fmt, ...)
+        self:_log(nil, S_INFO, fmt, ...)
+    end,
+    debug = function(self, fmt, ...)
+        self:_log(nil, S_DEBUG, fmt, ...)
+    end,
+    error = function(self, fmt, ...)
+        self:_log(nil, S_ERROR, fmt, ...)
+        if self.backtrace_on_error then self:_trace(S_ERROR) end
+    end,
+    syserror = function(self, fmt, ...)
+        self:_log(nil, S_SYSERROR, fmt, ...)
+        if self.backtrace_on_error then self:_trace(S_SYSERROR) end
+    end,
+    trace = function(self, log_level, level)
+        level = level or 1
+        if type(log_level) == 'string' then
+            log_level = log_level_name[log_level]
+            if log_level == nil then
+                error('bad argument #1 (log level expected)', 2)
+            end
+        elseif type(log_level) ~= 'number' then
+            log_level = S_ERROR
+        end
+        self:_trace(log_level, level)
+    end
+}
 
-local function log_rotate()
-    ffi.C.say_logrotate(0)
-end
+-- global table with all logger objects
+local logger_object_list = {}
 
-local function log_level(level)
-    if type(level) == 'string' then
-        level = log_level_name[level]
-        if level == nil then
-            error('bad argument #1 (log level expected)', 2)
+local function logger_object_new(options)
+    options = options or {}
+    -- prepare logger name
+    local name = options.name
+    if name == nil then
+        name = debug.getinfo(2).source
+        if name == nil then
+            name = 'internal'
+        else
+            name = fio.basename(name)
         end
     end
+    -- find and return logger, if it exists
+    local logger_object = logger_object_list[name]
```

Comment:
Nope, it's not. But we may once init object in the init.lua and use it everywhere in the code. simple and convenient